### ISSUE TITLE
Add tests to ensure compatibility with Moq 4.16

### ISSUE THREAD

**Issue by rjmurillo** [rjmurillo/moq.analyzers#420]
## Problem
`SetupShouldNotIncludeAsyncResultAnalyzer` skips diagnostics when the referenced Moq assembly is version 4.16 or newer. However, the original tests only covered older versions. Without a test case for Moq ≥4.16, regressions might cause the analyzer to erroneously report `Moq1201` when using modern Moq.

## Why this matters
Lack of coverage for the new Moq behavior means future changes could unintentionally break compatibility with Moq 4.16+, yet tests would still pass. Ensuring both old and new versions are exercised guards against such regressions.

```diff
 using Verifier = Moq.Analyzers.Test.Helpers.AnalyzerVerifier<Moq.Analyzers.SetupShouldNotIncludeAsyncResultAnalyzer>;
 
 namespace Moq.Analyzers.Test;
 
 public class SetupShouldNotIncludeAsyncResultAnalyzerTests(ITestOutputHelper output)
 {
     public static IEnumerable<object[]> TestData()
     {
-        return new object[][]
+        IEnumerable<object[]> old = new object[][]
         {
             ["""new Mock<AsyncClient>().Setup(c => c.TaskAsync());"""],
 
             // Prior to Moq 4.16, the setup helper ReturnsAsync(), ThrowsAsync() are preferred
             ["""new Mock<AsyncClient>().Setup(c => c.GenericTaskAsync()).ReturnsAsync(string.Empty);"""],
 
             // Starting with Moq 4.16, you can simply .Setup the returned tasks's .Result property
             ["""new Mock<AsyncClient>().Setup(c => {|Moq1201:c.GenericTaskAsync().Result|});"""],
         }.WithNamespaces().WithOldMoqReferenceAssemblyGroups();
+
+        IEnumerable<object[]> @new = new object[][]
+        {
+            // Moq 4.16 and later allows setting up Task<T>.Result directly
+            ["""new Mock<AsyncClient>().Setup(c => c.GenericTaskAsync().Result);"""],
+        }.WithNamespaces().WithNewMoqReferenceAssemblyGroups();
+
+        return old.Union(@new);
     }
 
     [Theory]
     [MemberData(nameof(TestData))]
     public async Task ShouldAnalyzeSetupForAsyncResult(string referenceAssemblyGroup, string @namespace, string mock)
     {
         string source =
             $$"""
               {{@namespace}}
 
               public class AsyncClient
               {
                   public virtual Task TaskAsync() => Task.CompletedTask;
 
                   public virtual Task<string> GenericTaskAsync() => Task.FromResult(string.Empty);
               }
 
               internal class UnitTest
               {
                   private void Test()
                   {
                       {{mock}}
                   }
               }
               """;
```

## Proposed fix
Extend the test data in `SetupShouldNotIncludeAsyncResultAnalyzerTests` to include a dataset using `WithNewMoqReferenceAssemblyGroups()`, verifying no diagnostic occurs when setting up `Task<T>.Result` with Moq 4.16+. The file currently shows this union of old and new data sets (lines 7‑26 above). The existing `[Theory]` remains valid for both data sets.

Running dotnet test after making this change will confirm that the analyzer behaves as intended for all supported Moq versions.
